### PR TITLE
Textfield component

### PR DESCRIPTION
**Pull request by wakooka**

### CODE REVIEW

File: src/autoComplete/index.unit.tsx [hunk 1]
```diff
@@ -343,18 +343,18 @@ describe('AutoComplete', () => {
 
   describe('#focus', () => {
     it('Can add focus to TextField', () => {
-      const wrapper = shallow(<AutoComplete {...defaultProps} focus />)
+      const wrapper = mount(<AutoComplete {...defaultProps} focus />)
```
Comment (pith) [inline src/autoComplete/index.unit.tsx]: You need to update the `AutoComplete` import that is why your test fails

File: src/autoComplete/index.unit.tsx [hunk 2]
```diff
@@ -343,18 +343,18 @@ describe('AutoComplete', () => {
...
       expect(wrapper.find('TextField').prop('focus')).toBe(true)
     })
     it('Should not add border focus on Textfield', () => {
-      const wrapper = shallow(<AutoComplete {...defaultProps} />)
+      const wrapper = mount(<AutoComplete {...defaultProps} />)
```
Comment (pith) [inline src/autoComplete/index.unit.tsx]: Why `mount` ?

File: src/textField/Textfield.tsx [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+type errorField = string | JSX.Element
+
+export interface TextFieldProps extends CommonFormFields {
+  isTextArea?: boolean
+  defaultValue?: string
+  labelledBy?: string
+  onChange?: (obj: OnChangeParameters) => void
+  onClear?: () => void
+  className?: Classcat.Class
```
Comment (pith) [inline src/textField/Textfield.tsx]: You forgot to add the `extraClassName`.

File: src/autoComplete/index.unit.tsx (hunk 1 above)
Comment (wakooka) [reply src/autoComplete/index.unit.tsx]: I can't, it's imported from the component Autocomplete itself, it's not imported in the test

File: src/autoComplete/index.unit.tsx (hunk 2 above)
Comment (wakooka) [reply src/autoComplete/index.unit.tsx]: Because wrapper.find('TextField') can only work with mount since now it's wrapped in the Styled HOC

**Comment by Tuxosaurus** [comment]
Rename files with "Textfield" to "TextField"

File: src/textField/Textfield.tsx (hunk 3 above)
Comment (Tuxosaurus) [reply src/textField/Textfield.tsx]: Not needed anymore :)

Comment (wakooka) [reply src/textField/Textfield.tsx]: Removed :)

**Comment by wakooka** [comment]
PTAL

**Pull request merged**